extensions/resteasy-reactive/rest-client-reactive/deployment/src/test/java/io/quarkus/rest/client/reactive/beanparam/BeanFormParamTest.java

```diff
@@ -0,0 +1,73 @@
+package io.quarkus.rest.client.reactive.beanparam;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.net.URI;
+
+import javax.ws.rs.BeanParam;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.FormParam;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.core.MediaType;
+
+import org.eclipse.microprofile.rest.client.RestClientBuilder;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.test.QuarkusUnitTest;
+import io.quarkus.test.common.http.TestHTTPResource;
+
+public class BeanFormParamTest {
+    @RegisterExtension
+    static final QuarkusUnitTest TEST = new QuarkusUnitTest();
+
+    @TestHTTPResource
+    URI baseUri;
+
+    @Test
+    void shouldPassFormParamsFromBeanParam() {
+        assertThat(clientBuilder().build(FormTestClient.class).postFormParams(new BeanWithFormParams("value1", "value2")))
+                .isEqualTo(
+                        "received value1-value2");
+    }
+
+    private RestClientBuilder clientBuilder() {
+        return RestClientBuilder.newBuilder().baseUri(baseUri);
+    }
+
+    @Path("/form")
+    public interface FormTestClient {
+        @POST
+        @Consumes(MediaType.APPLICATION_FORM_URLENCODED)
+        String postFormParams(@BeanParam BeanWithFormParams beanParam);
+    }
+
+    public static class BeanWithFormParams {
+        private final String param1;
+        private final String param2;
+
+        public BeanWithFormParams(String param1, String param2) {
+            this.param1 = param1;
+            this.param2 = param2;
+        }
+
+        @FormParam("param1")
+        public String getParam1() {
```

Comment:
could you make one of the params use a converter?

Reply:
Good idea, I will add a corresponding text in the next few hours

Reply:
done